core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -331,4 +383,128 @@ private static PartitionData toPartitionData(StructLike key, Types.StructType ke
       return keyTemplate.copyFor(key);
     }
   }
+
+  /** Distributed scan for partitions table that creates multiple tasks for processing manifests. */
+  private static class DistributedPartitionsScan extends BaseMetadataTableScan {
```

Comment:
I feel like we should model the distributed partitions table scan after the distributed scans for other metadata tables, and have logic in the `spark` module where needed instead of here.